src/mobot_client/payments/client.py

```diff
@@ -53,34 +54,38 @@ def check_receiver_receipt_status(self, address, receipt):
             self.logger.exception("Exception raised when checking receipt")
             raise CheckReceiptException(str(e))
 
-    def _wait_for_transaction(self, receipt) -> PaymentStatus:
+    def _wait_for_transaction(self, payment: Payment) -> Payment:
+        receipt = payment.signal_payment.receipt
         transaction_status = PaymentStatus.PENDING
-        txo_id = None
         while transaction_status == PaymentStatus.PENDING:
             receipt_status = self.check_receiver_receipt_status(
                 self.public_address, receipt
             )
             transaction_status = PaymentStatus[receipt_status["receipt_transaction_status"]]
-            txo_id = receipt_status["txo"]["txo_id"]
             self.logger.info(f"Waiting for {receipt}, current status {receipt_status}")
+            payment.status = transaction_status
+            payment.save()
             time.sleep(2)
 
         if transaction_status != PaymentStatus.SUCCESS:
             self.logger.error(f"failed {transaction_status}")
             transaction_status = PaymentStatus.FAILURE
+            payment.status = transaction_status
+            payment.save()
 
-        return txo_id, transaction_status
+        return payment
```

Comment:
It looks like the return type was changed here, but code consuming this function has remained the same elsewhere in this commit (at least here, but also perhaps elsewhere: https://github.com/mobilecoinofficial/mobot/pull/72/commits/fe35caa2ec6a645200550701bfcee64fd34e7531#diff-21df8d68dc58df1edff1f0784b711a9889a48dac8b963ad4541278d7e53543ceR81) 🤔

Reply:
Ohhh do we not do atomic commits? I would love to encourage us to do that when the time is right!

Noted for now though, carry on 🚀